src/main/java/com/ft/methodearticlemapper/transformation/RecommendedXMLEventHandler.java

```diff
@@ -0,0 +1,119 @@
+                                        BodyProcessingContext bodyProcessingContext) throws XMLStreamException {
+        if (isElementOfCorrectType(startElement)) {
+            RecommendedData recommendedData = recommendedXMLParser.parseElementData(
+                    startElement, xmlEventReader, bodyProcessingContext);
+            if (recommendedData.getLinks().size() > 0) {
+                eventWriter.writeStartTag(RECOMMENDED_TAG, noAttributes());
+                writeRecommendedTitle(eventWriter, recommendedData);
+                writeIntro(eventWriter, recommendedData);
+                writeList(eventWriter, recommendedData);
+                eventWriter.writeEndTag(RECOMMENDED_TAG);
+            }
+        } else {
+            throw new BodyProcessingException("Expected event start element:" + RECOMMENDED_TAG + ". Received start element: " + startElement.getName().getLocalPart());
+        }
+    }
+
+    private void writeList(BodyWriter eventWriter, RecommendedData recommendedData) {
+        eventWriter.writeStartTag(LIST_TAG, noAttributes());
+        for (RecommendedData.Link link : recommendedData.getLinks()) {
+            eventWriter.writeStartTag(LIST_ITEM_TAG, noAttributes());
+            writeAnchor(eventWriter, link.title, getAnchorAttributes(link));
+            eventWriter.writeEndTag(LIST_ITEM_TAG);
+        }
+        eventWriter.writeEndTag(LIST_TAG);
+    }
+
+    private Map<String, String> getAnchorAttributes(RecommendedData.Link link) {
+        Matcher matcher = pattern.matcher(link.address);
+        Map<String, String> attributes = new HashMap<>();
+        if (matcher.find()) {
+            attributes.put(TYPE_ATTRIBUTE, TYPE_VALUE);
+            attributes.put(URL_ATTRIBUTE, BASE_URL + matcher.group(2));
+        } else {
+            attributes.put(HREF_ATTRIBUTE, link.address);
+        }
+        return attributes;
+    }
+
+    private boolean isElementOfCorrectType(StartElement event) {
+        return event.getName().getLocalPart().toLowerCase().equals(RECOMMENDED_TAG);
+    }
+
+    private Map<String, String> noAttributes() {
+        return Collections.emptyMap();
+    }
+
+
+    private void writeRecommendedTitle(BodyWriter eventWriter, RecommendedData recommendedData) {
+        writeElement(eventWriter, recommendedData.getTitle(), RECOMMENDED_TITLE_TAG, noAttributes());
+    }
+
+    private void writeIntro(BodyWriter eventWriter, RecommendedData recommendedData) {
+        writeElementIfDataNotEmpty(eventWriter, recommendedData.getIntro(), INTRO_TAG, noAttributes());
+    }
+
+    private void writeAnchor(BodyWriter eventWriter, String title, Map<String, String> attributes) {
+        writeElement(eventWriter, title, ANCHOR_TAG, attributes);
+    }
+
+    private void writeElement(BodyWriter eventWriter, String dataField, String elementName, Map<String, String> attributes) {
+        eventWriter.writeStartTag(elementName, attributes);
+        if (StringUtils.isNotEmpty(dataField)) {
```

Comment:
The title could be missing but there will be a href attribute that will be added to the anchor tag. That href attribute can't be empty or missing because I make sure of that in the RecommendedXMLParser.